start.py

```diff
@@ -0,0 +1,275 @@
+CLUSTERDOCK_CLIENT_CONTAINER_DIR = '/etc/clusterdock'
+
+TLS_CONF_URL = 'https://pulsar.incubator.apache.org/docs/latest/security/openssl.cnf'
+TLS_DIR = '{}/pulsar_tls'.format(CLUSTERDOCK_CLIENT_CONTAINER_DIR)
+TLS_CLIENT_DIR = '{}/client'.format(TLS_DIR)
+
+logger = logging.getLogger('clusterdock.{}'.format(__name__))
+
+
+def main(args):
+    quiet = not args.verbose
+
+    node_image = '{}/{}/topology_pulsar:pulsar-{}'.format(args.registry,
+                                                          args.namespace or DEFAULT_NAMESPACE,
+                                                          args.pulsar_version)
+    ports = [{WEB_SERVICE_PORT: WEB_SERVICE_PORT} if args.predictable else WEB_SERVICE_PORT,
+             {WEB_SERVICE_TLS_PORT: WEB_SERVICE_TLS_PORT} if args.predictable else WEB_SERVICE_TLS_PORT,
+             {BROKER_SERVICE_PORT: BROKER_SERVICE_PORT} if args.predictable else BROKER_SERVICE_PORT,
+             {BROKER_SERVICE_TLS_PORT: BROKER_SERVICE_TLS_PORT} if args.predictable else BROKER_SERVICE_TLS_PORT]
+
+    clusterdock_config_host_dir = os.path.realpath(os.path.expanduser(args.clusterdock_config_directory))
+    volumes = [{clusterdock_config_host_dir: CLUSTERDOCK_CLIENT_CONTAINER_DIR}]
+
+    proxy_node = Node(hostname=args.proxy_node_name,
+                      group='proxy',
+                      image=node_image,
+                      ports=ports,
+                      volumes=volumes)
+    broker_nodes = [Node(hostname=hostname, group='broker', image=node_image, volumes=volumes)
+                    for hostname in args.broker_nodes]
+    zk_nodes_to_start = [Node(hostname=hostname, group='zookeeper', image=node_image, volumes=volumes)
+                         for hostname in args.zookeeper_nodes if hostname != proxy_node.hostname]
+    nodes = [proxy_node] + broker_nodes + zk_nodes_to_start
+    cluster = Cluster(*nodes)
+    cluster.start(args.network)
+    # since zk node can be on a proxy, check and add in its list
+    zk_nodes = zk_nodes_to_start + [node for node in [proxy_node] if node.hostname in args.zookeeper_nodes]
+
+    # zookeeper
+    for idx, node in enumerate(zk_nodes, start=1):
+        zookeeper_conf = node.get_file(ZOOKEEPER_CONF)
+        zookeeper_properties = PropertiesFile.loads(zookeeper_conf)
+        for srvidx, srvnode in enumerate(zk_nodes, start=1):
+            zookeeper_properties['server.{}'.format(srvidx)] = '{}.{}:2888:3888'.format(srvnode.hostname,
+                                                                                        cluster.network)
+        node.put_file(ZOOKEEPER_CONF, PropertiesFile.dumps(zookeeper_properties))
+        zookeeper_commands = [
+            'mkdir -p {}/data/zookeeper'.format(PULSAR_HOME),
+            'echo {} > {}/data/zookeeper/myid'.format(idx, PULSAR_HOME),
+            '{}/bin/pulsar-daemon start zookeeper'.format(PULSAR_HOME)
+        ]
+        node.execute('; '.join(zookeeper_commands), quiet=quiet)
```

Comment:
Might wanna separate these with `&&` to fail fast in case any of the commands fail in the process (here and elsewhere).